src/content/blog/2025-01-31-ilp-packet-lifecycle.mdx

```diff
@@ -0,0 +1,96 @@
+### Responding with Fulfill or Reject
+In the second phase of STREAM's execution process, the receiver will need to handle each Prepare packet with either a fulfillment or a rejection, so that the sender knows whether or not a given Prepare packet was successfully sent or not.
+The receiver fulfills a Prepare packet by sending a [Fulfill packet](/developers/rfcs/interledger-protocol/) back to the sender with the preimage from which the Prepare packet's condition was generated.
+
+![ILP Fulfill Packet Specification](/developers/img/blog/2025-01-31/ilp-fulfill-spec.png)
+
+Since only the receiver can provide the preimage needed to fulfill a "prepare" packet's condition, no node in the network in between it and the sender can intercept the packet and send back a fulfilled packet without it reaching the receiver first.
+As soon as the sender receives enough fulfillments from the receiver to complete the payment, it stops sending packets over STREAM and closes the connection.
+
+If the received Prepare packet expired before it reached the receiver, the amount received already from prior packets exceeds the total, or some other error occurs, the receiver will instead respond with a Reject packet instead.
+
+
+On another technical point, there also exist some types of packets that cannot be fulfilled. For instance, an unfulfillable packet gets sent when an Open Payments quote is created. This packet's purpose is to determine what the total cost of the payment will be by factoring any fees or conversion rates that may be applied, instead of trying to fulfill a payment.
+
+### Wrapping Things Up
+As "fulfill" packets arrive at the sender, it verifies that the preimage sent back in each of them does indeed fulfill the condition sent with the corresponding "prepare" packet.
+Once the STREAM connection finishes sending all of the packets and verifies the corresponding fulfillments from the receiver, it closes. The sending Rafiki instance will then update its outgoing payment as being complete, and the receiving Rafiki instance publishes a webhook to its account servicing entity directing it to mark its incoming payment as complete as well.
+
+## Conclusion
+The Interledger Protocol had to go through a number of important changes in order to position itself as a future-proof way to send money. Packetizing Interledger payments was a significant change that brought it in line with the design patterns of the modern web that it seeks to be a part of.
+These packets continue to play this role as a part of Rafiki, as it coordinates those payments with Open Payments and manages the ILP connections used to fulfill them.
+
+If this post has established the connection between Rafiki, Open Payments, and Interledger in a way that piques interest, consider delving futher into these Interledger concepts:
+
+- [ILP v4](/developers/rfcs/interledger-protocol/)
+- [STREAM](/developers/rfcs/stream-protocol/)
+- [Connectors & Peering](/developers/rfcs/peering-clearing-settling)
```

Comment:
As the ending can you please add the footer from https://interledger.org/developers/blog/rafiki-tigerbeetle-integration/

_As we are open source, you can easily check our work on [GitHub](https://github.com/interledger/). If the work mentioned here inspired you, we welcome your contributions. You can join our [community slack](https://communityinviter.com/apps/interledger/interledger-working-groups-slack) or participate in the next [community call](https://docs.google.com/document/d/1T_Q_eRZtL8GluJR9rVADd7fvkB8mPluMk-d96LM4vlg/edit?usp=sharing), which takes place each second Wednesday of the month.

If you want to stay updated with all open opportunities and news from the Interledger Foundation, you can subscribe to our [newsletter](https://interledger.org/subscribe)._

Reply:
I think the footer is added automatically. I see it on the dev site when I launch it locally, and the Tigerbeetle post doesn't seem to add anything extra to include the footer [in the source](https://github.com/interledger/interledger.org-developers/blob/main/src/content/blog/2024-12-17-rafiki-tigerbeetle-integration.mdx?plain=1).

Reply:
@ioana-chiorean they are all added automatically now. So we shouldn't repeat it.